nautobot/ipam/factory.py

```diff
@@ -304,3 +361,184 @@ def _adjust_kwargs(cls, **kwargs):
+    is_pool = factory.Faker("pybool")
+    # TODO: create a LocationGetOrCreateFactory to get or create a location with matching site
+    location = factory.Maybe("has_location", random_instance(lambda: Location.objects.get_for_model(Prefix)), None)
+    role = factory.Maybe("has_role", random_instance(Role), None)
+    # TODO: create a SiteGetOrCreateFactory to get or create a site with matching tenant
+    site = factory.Maybe("has_site", random_instance(Site), None)
+    status = factory.Maybe(
+        "is_container",
+        factory.LazyFunction(lambda: Prefix.STATUS_CONTAINER),
+        random_instance(
+            lambda: Status.objects.get_for_model(Prefix).exclude(pk=Prefix.STATUS_CONTAINER.pk), allow_null=False
+        ),
+    )
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant))
+    vlan = factory.Maybe(
+        "has_vlan",
+        factory.SubFactory(
+            VLANGetOrCreateFactory,
+            location=factory.SelfAttribute("..location"),
+            site=factory.SelfAttribute("..site"),
+            tenant=factory.SelfAttribute("..tenant"),
+        ),
+        None,
+    )
+    vrf = factory.Maybe(
+        "has_vrf",
+        factory.SubFactory(VRFGetOrCreateFactory, tenant=factory.SelfAttribute("..tenant")),
+        None,
+    )
+
+    @factory.post_generation
+    def children(self, create, extracted, **kwargs):
+        """
+        Create child prefixes and ip addresses within the prefix IP space. Defaults to generating
+        0-4 child prefixes for parents that are containers, or generating 0-4 ip addresses within
+        the prefix's IP space for non-container prefixes. Only creates children on 50% of prefixes.
+        Set children__max_count to an integer when calling the factory creation methods
+        (`create()`, `create_batch()`, etc) to override the maximum number of children generated.
+        """
+        if extracted:
+            # Objects have already been created, do nothing
+            return
+
+        # 50% chance to create children
+        if not faker.Faker().pybool():
+            return
+
+        action = "create" if create else "build"
+        is_ipv6 = self.family == 6
+
+        # Create child prefixes for containers, otherwise create child ip addresses
+        factory = PrefixFactory if self.status == Prefix.STATUS_CONTAINER else IPAddressFactory
+        method = getattr(factory, action)
+
+        # Default to maximum of 4 children unless overridden in kwargs
+        max_count = int(kwargs.pop("max_count", 4))
+        child_count = faker.Faker().pyint(min_value=0, max_value=min(max_count, self.prefix.size))
+        if child_count == 0:
+            return
+
+        if factory == IPAddressFactory:
+            # Create child ip addresses, preserving vrf, tenant and is_ipv6 from parent
+            for count, address in enumerate(self.prefix.iter_hosts()):
+                if count == child_count:
+                    break
```

Comment:
This will always give us a contiguous block of allocated IPs, right? Wonder if we could add some coin-flip logic to randomly skip ahead, something like:
```suggestion
            count = 0
            for address in self.prefix.iter_hosts():
                if not faker.Faker().pybool():
                    continue
                count += 1
                if count == child_count:
                    break
```